lib/io.toit

```diff
@@ -1233,31 +1282,24 @@ class Buffer extends CloseableWriter:
     buffer.big-endian.write-int32 0x12345678
     // The most significant byte 0x12 is at index 0.
     print buffer.bytes  // => #[0x12, 0x34, 0x56, 0x78]
-  ```
-
-  ```
-  import io
 
-  main:
-    buffer := io.Buffer
-    writer := buffer.big-endian
-    writer.write "Can be used like a normal writer."
-    writer.write-int32 0x12345678
-    result := buffer.bytes
-    ...
+    // The buffer version also supports 'put' operations
+    buffer.big-endian.put-int32 --at=0 0x11223344
   ```
   */
   big-endian -> EndianBuffer:
-    return EndianBuffer --buffer=this --byte-order=binary.BIG_ENDIAN
+    if not endian_ or endian_.byte-order_ != binary.BIG_ENDIAN:
```

Comment:
I wouldn't read the field that many times.

Reply:
done.